src/spasht/ui/ui-spasht.php

```diff
@@ -141,6 +141,12 @@ public function Output()
     $typeName = trim(GetParm("typeName", PARM_STRING));
     $providerName = trim(GetParm("providerName", PARM_STRING));
 
+    // Reading contribution data
+    $coordinatesInfo = trim(GetParm("jsonStringCoordinates", PARM_STRING));
```

Comment:
The coordinates should be read from the database for the upload. A client can manipulate the request data and put invalid coordinates.